ClothesApp/WebAPI/Controllers/OrderController.cs

```diff
@@ -0,0 +1,81 @@
+using Application.Dtos.OrderItems;
+using Application.Dtos.Orders;
+using Application.Dtos.OrderTransactions;
+using Application.Interfaces.Services;
+using Domain.Entities;
+using Domain.Enums;
+using Microsoft.AspNetCore.Mvc;
+
+namespace WebAPI.Controllers;
+
+[ApiController]
+[Route("api/orders")] 
+public class OrderController : ControllerBase
+{
+    private readonly IOrderService _orderService;   
+    
+    public OrderController(IOrderService orderService)
+    {
+        _orderService = orderService;
+    }
+    
+    [HttpGet("{id}/history")]
+    [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(IList<OrderTransactionsDto>))]
+    [ProducesResponseType(StatusCodes.Status404NotFound, Type = typeof(ProblemDetails))]
+    public async Task<ActionResult<IList<OrderDto>>> GetOrderHistory([FromRoute] long id)
+    {
+        var orderHistory = await _orderService.GetOrderHistoryByOrderId(id);
+
+        return Ok(orderHistory);
+    }
+    
+    [HttpPost]
+    [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(OrderDto))]
+    [ProducesResponseType(StatusCodes.Status404NotFound, Type = typeof(ProblemDetails))]
+    public async Task<ActionResult<IList<OrderDto>>> CreateOrderForCustomer([FromBody] OrderInputDto orderInputDto)
+    {
+        var orderDto = await _orderService.Add(orderInputDto);
+
+        return Ok(orderDto);
+    }
+    
+    [HttpPost("{id}/submit")]
+    [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(OrderDto))]
+    [ProducesResponseType(StatusCodes.Status404NotFound, Type = typeof(ProblemDetails))]
+    public async Task<ActionResult<IList<OrderDto>>> SubmitOrder([FromRoute] long id)
```

Comment:
What's the meaning of submit order method?

Reply:
It's change status from OnHold to InReview. But we can store order items in cookies and set InReview on PostOrder endpoint